tests/python/unittest/test_higher_order_grad.py

```diff
@@ -129,6 +131,44 @@ def check_second_order_unary(x, op, grad_grad_op):
     # Validate the gradients.
     assert_almost_equal(expected_grad_grad, x.grad.asnumpy())
 
+class RandomShapes(object):
+    def __init__(self, dim):
+        self.dim = dim
+        self.curdim = 1
+
+    def __iter__(self):
+        return self
+
+    def next(self):
+        return self.__next__()
+
+    def __next__(self):
+        if self.curdim > self.dim:
+            raise StopIteration
+        shape = rand_shape_nd(self.curdim)
+        print(shape)
+        x = nd.random.normal(shape=shape)
+        self.curdim += 1
+        return x
+
+
+@with_seed()
+def test_dense_backward():
```

Comment:
Can we add a test for FullyConnected operator itself?

Reply:
Dense calls FullyConnected during forward, do you suggest to use the symbolic API?

Reply:
Since your change in C++ is directly on the FullyConnected operator, wouldn't it make sense to test on it directly?

Reply:
Makes sense, but for convenience I prefer to test with Gluon. I don't think is worth the complexity and extra work now. Let me know if you disagree.

Reply:
The reason I was suggesting to test FullyConnected operator itself is then you could reuse the `check_second_order_unary` utility methods like all the other operators do. Having to write ad hoc test for each operator is not scalable and error prone.